main.tf

```
@@ -1 +1,57 @@
 
+resource "azurerm_mysql_flexible_server" "this" {
+  name                         = var.server_name
+  location                     = var.location
+  resource_group_name          = var.resource_group_name
+  version                      = var.server_version
+  administrator_login          = var.administrator_login
+  administrator_password       = var.administrator_password
+  sku_name                     = var.sku_name
+  geo_redundant_backup_enabled = var.geo_redundant_backup_enabled
+  zone                         = var.zone
+
+  storage {
+    auto_grow_enabled = var.auto_grow_enabled
+    size_gb           = var.size_gb
```

Should probably prefix these variables with the block name:

```suggestion
    auto_grow_enabled = var.storage_auto_grow_enabled
    size_gb           = var.storage_size_gb
```